src/backend/distributed/connection/connection_management.c

```diff
@@ -386,6 +345,28 @@ FindAvailableConnection(dlist_head *connections, uint32 flags)
 			continue;
 		}
 
+		if ((flags & NO_DATA_ACCESS_CONNECTION) != 0)
+		{
+			if (!connection->isNonDataAccess &&
```

Comment:
Should this be `||` instead of `&&` ?

Reply:
No, but it does seem slightly unintuitive. Maybe I should swap the if-else.

Reply:
So If a connection is used for data access but it is not used for any placements, we want to use that connection for non data access query?

Or If a connection is non data access, but it is used for some placement, do we want to return that connection?